omod/src/main/java/org/openmrs/module/smartonfhir/page/controller/FindPatientPageController.java

```diff
@@ -24,9 +24,10 @@
 public class FindPatientPageController {
 	
 	public void get(PageModel model, @RequestParam("app") AppDescriptor app, @RequestParam("token") String token,
-	        UiSessionContext sessionContext, UiUtils ui) throws UnsupportedEncodingException {
+	        @RequestParam("launchType") String launchType, UiSessionContext sessionContext, UiUtils ui)
+	        throws UnsupportedEncodingException {
 		model.addAttribute("afterSelectedUrl", app.getConfig().get("afterSelectedUrl").getTextValue() + "&token="
-		        + URLEncoder.encode(token, StandardCharsets.UTF_8.name()));
+		        + URLEncoder.encode(token, StandardCharsets.UTF_8.name()) + "&launchType=" + launchType);
```

Comment:
Please URL encode the `launchType` as we do with the `token`

Reply:
Thanks, I will encode this too.